### PR TITLE
SaveState: Keep an undo for each slot by default

### PR DESCRIPTION
**Pull request by unknownbrackets**
No real UI for it ATM, but at least there's a safety mechanism.  Otherwise you just find out you hit the wrong key or made a horrible mistake.

At least this way, you might find it or ask online.

-[Unknown]

### CODE REVIEW

**Comment by hrydgard** [comment]
It's a nice feature but I'm worried about unexpected disk utilization. A lot of people run their Android devices pretty full, and doubling the size of each save slot makes the impact pretty large, and without UI, it's not so easy for many of those to even make use of it. Maybe only default it to on for PC users?

**Comment by unknownbrackets** [comment]
Right, makes sense.  I added to to the System menu so it could still be enabled easily on mobile.  More reason people might know it exists.

-[Unknown]

File: Core/SaveState.cpp [hunk 1]
```diff
@@ -433,12 +457,43 @@ namespace SaveState
 		}
 	}
 
+	bool UndoSaveSlot(const std::string &gameFilename, int slot) {
```
Comment (hrydgard) [inline Core/SaveState.cpp]: Nothing calls this function, I guess it's here for future UI?

Comment (unknownbrackets) [reply Core/SaveState.cpp]: Right, I just had trouble putting a button in a good place that didn't seem ugly or annoying.

-[Unknown]

**Pull request merged**